### PR TITLE
PA-1712: Update logic to use pod status instead of ns

### PR DESCRIPTION
**Pull request by kshithijiyer-px**
**What this PR does / why we need it**:

- [x] Rename `checkBackupObjectForNonExpectedNS` to `checkBackupObjectForUnexpectedNS`
- [x] Update logic to use pod status instead of ns

https://portworx.atlassian.net/browse/PB-4632 needs to be fixed to run this fix. 

**Which issue(s) this PR fixes** (optional)
Closes #PA-1712

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by trenukarya-px** [comment]
@kshithijiyer-px Please add more details why to change to POD instead of NS. Can't make out from the ticket number provided.

**Comment by kshithijiyer-px** [comment]
> @kshithijiyer-px Please add more details why to change to POD instead of NS. Can't make out from the ticket number provided.

When we do a replace it usually cleans up the namespace but doesn't delete the namespace itself. The final output of this would be that the pods and pvcs would be recreated. Hence we need to check for pods age instead of ns age.

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -5140,37 +5140,43 @@ func checkBackupObjectForNonExpectedNS(ctx context.Context, backupName string) e
 	return nil
 }
 
-// getNamespaceAge gets the namespace age of all the namespaces on the cluster
-func getNamespaceAge() (map[string]time.Time, error) {
-	var namespaceAge = make(map[string]time.Time)
+// getPodAge gets the pod age of all pods on all the namespaces on the cluster
+func getPodAge() (map[string]time.Time, error) {
```
Comment (kphalgun-px) [inline tests/backup/backup_helper.go:5144]: I feel map[string][string]time.Time (map[namespace][pod]time.Time) would be more useful.

**Review by kphalgun-px — APPROVED** [review]
LGTM

File: tests/backup/backup_helper.go (hunk 1 above)
Comment (kshithijiyer-px) [reply tests/backup/backup_helper.go]: Nice catch it would be much cleaner. It would slightly increase complexity but approach is much cleaner.

**Comment by kshithijiyer-px** [comment]
> LGTM. Except for one minor nit. Can we rename the func `checkBackupObjectForNonExpectedNS` to `checkBackupObjectForUnexpectedNS`

Done

**Pull request merged**